### PR TITLE
Improve date handling in the EndNote plugin

### LINKED ISSUE #46: Import failure due to runtime errors parsing dates
When you try to import a deduplicated search from EndNote to Rayyan, the import fails. 

Customer says: I have tried the xml, txt and csv formats, but some error or other keeps coming up. Either "Wrong number of arguments (5, expecting 1-3)" or the most common one is "Undefined method 'sub'..".

Some references have the date like this:
%8 Sep 2011
2022-04-27
More failing examples are available upon request.

This will throw the following error:
Error: wrong number of arguments (given 5, expected 1..3)

### PR DESCRIPTION
**Pull request by kohrVid**
Closes #46 

Following an issue reported by a user, this PR refactors the `.date_array` method used to import dates in EndNote files. Currently, errors are thrown when duplicate date fields are imported in a single article (as these are treated as an array of dates in a single field by the parser) and no date is imported if two dates appear in the same field. This commit assumes that the last date entered in these scenarios is the correct one.

This PR does not fix issues that are the result of articles imported without an empty line in between records. This issue relates to a [wider problem with the parser](https://github.com/rayyansys/ref_parsers/issues/9) and so would need to be addressed separately.

### CODE REVIEW

File: lib/rayyan-formats-plugins/plugins/endnote.rb [hunk 1]
```diff
@@ -94,6 +94,9 @@ def get_date_array(article)
         def date_array(date)
           return [] if date.nil? || date.empty?
 
+          date = date.last if date.is_a?(Array)
+          date = date&.split(/\s{2,}/)&.last if date.is_a?(String)
```
Comment (hammady) [inline lib/rayyan-formats-plugins/plugins/endnote.rb:98]: Can you please explain what this line covers?

Comment (kohrVid) [reply lib/rayyan-formats-plugins/plugins/endnote.rb:98]: Sure. Line 97 covers instances where there are two or more `%D` or `%8` fields in a single article.
Line 98 covers something like the example in the issue that you raised where the date might look like this:

    %8 Sep 2011
    2022-04-27

The parser treats the example above as a single line with two dates separated by spaces. My code assumes that the last date (e.g., "2022-04-27")  is the correct one. What do you think about this change? I was having trouble deciding on whether handling these dates might be overkill - is it still worth trying to handle dates that appear in this format?

If you think this code's worth keeping, I can add some inline comments to make it clearer a little later

Comment (hammady) [reply lib/rayyan-formats-plugins/plugins/endnote.rb:98]: Are you referring to the line merger defined [here](https://github.com/rayyansys/ref_parsers/blob/master/lib/ref_parsers/line_parser.rb#L2)? If that's the case, I believe we should use that constant. Moreover, splitting over that constant exact match and getting the last item in the array would be my choice.

Comment (kohrVid) [reply lib/rayyan-formats-plugins/plugins/endnote.rb]: Thanks - had no idea about the constant before you mentioned it! :sweat_smile:  Have updated the PR to use that instead

Comment (hammady) [reply lib/rayyan-formats-plugins/plugins/endnote.rb]: Sorry bad documentation on the `ref_parsers` side.

**Review by hammady — APPROVED** [review]

**Pull request merged**